pkgs/development/python-modules/rapidocr-onnxruntime/default.nix

```
@@ -0,0 +1,107 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  fetchurl,
+  linkFarm,
+
+  pythonRelaxDepsHook,
+  pytestCheckHook,
+
+  setuptools,
+  pyclipper,
+  opencv4,
+  numpy,
+  six,
+  shapely,
+  pyyaml,
+  pillow,
+  onnxruntime,
+}:
+let
+  version = "1.3.22";
+
+  src = fetchFromGitHub {
+    owner = "RapidAI";
+    repo = "RapidOCR";
+    rev = "v${version}";
+    hash = "sha256-8h4DFhnI9imr+bYQZdlrl8UKUdpwnGK+SGxLXSMmcag=";
+  };
+
+  testRev = "38f67e0558416db6056ddd8613292d23b5ff002c";
+
+  testFiles = {
+    "PP-OCRv4/ch_PP-OCRv4_det_infer.onnx" = "sha256-0qdyDUWlQlcgix4T42qEeYlMt0FVpe/ilGJRLUL0nak=";
+    "PP-OCRv4/ch_PP-OCRv4_rec_infer.onnx" = "sha256-SPxA8k9tKiB6KxCR00N+s8w+trZ23D75w3OEAFSDaDs=";
+    "PP-OCRv1/ch_ppocr_mobile_v2.0_cls_infer.onnx" = "sha256-5HrO32YyMPiGP/GrDmTdLYK4OPzrWVcUbasYWonWIVw=";
+  };
+
+  testData = lib.pipe testFiles [
+    (lib.mapAttrsToList (name: hash: {
+      inherit name;
+      path = fetchurl {
+        url = "https://huggingface.co/SWHL/RapidOCR/resolve/${testRev}/${name}";
+        inherit hash;
+      };
+    }))
+    (linkFarm "rapidocr-onnxruntime-test-deps")
+  ];
+in
+buildPythonPackage {
+  pname = "rapidocr-onnxruntime";
+  inherit version src;
+  pyproject = true;
+
+  sourceRoot = "${src.name}/python";
+
+  # The setup.py supplied by upstream tries to determine the current version by
+  # fetching the latest version of the package from PyPI, and then bumping the version number.
+  # This is not allowed in the Nix build environment as we do not have internet access,
+  # hence we patch that out and get the version from the build environment directly.
+  patches = [ ./setup-py-override-version-checking.patch ];
+
+  nativeBuildInputs = [
+    pythonRelaxDepsHook
+    pytestCheckHook
+  ];
+
+  build-system = [ setuptools ];
+
```

```suggestion
  preBuild = ''
    mkdir rapidocr_onnxruntime_t
    mv rapidocr_onnxruntime rapidocr_onnxruntime_t
    mv rapidocr_onnxruntime_t rapidocr_onnxruntime
  '';
```
I don't know what they want to do.
`pythonImportsCheck` now passes, but it still seems to be missing some files during the test.

https://github.com/RapidAI/RapidOCR/blob/02829ef986bc2a5c4f33e9c45c9267bcf2d07a1d/.github/workflows/gen_whl_to_pypi_rapidocr_ort.yml#L83-L85

---

before
```shell
$ cat /tmp/nix-build-python3.11-rapidocr-onnxruntime-1.3.22.drv-2/source/python/rapidocr_onnxruntime/__init__.py 
from .rapidocr_onnxruntime.main import RapidOCR, VisRes
```

after
```shell
$ cat /tmp/nix-build-python3.11-rapidocr-onnxruntime-1.3.22.drv-3/source/python/rapidocr_onnxruntime/rapidocr_onnxruntime/__init__.py 
# -*- encoding: utf-8 -*-
# @Author: SWHL
# @Contact: liekkaskono@163.com
from .main import RapidOCR
from .utils import LoadImageError, VisRes
```